bindings/matrix-sdk-ffi/src/sliding_sync.rs

```diff
@@ -199,6 +199,10 @@ impl SlidingSyncRoom {
             .map(|room| Arc::new(Room::with_timeline(room, self.timeline.clone())))
     }
 
+    pub fn avatar_url(&self) -> Option<String> {
+        self.client.get_room(self.inner.room_id()).and_then(|room| room.avatar_url_string())
```

Comment:
```suggestion
        Some(self.client.get_room(self.inner.room_id())?.into())
```

`.into()` might still not be much as efficient as what you wrote before theoretically, but unless you can show us a benchmark where it actually makes a difference, I don't think we should be complicating the public API when the only benefit is to have fewer allocations in theory.

Reply:
I think I agree here, it's just a fact of biting the bullet. Although it's negligible, it still gives me an icky feeling :smile:. But I think this should be solved if `MxcUri` as a type gets exposed, as it could just be returned as-is.